server/app/core/migrator.py

```
@@ -0,0 +1,185 @@
+    logging.getLogger(__name__).info(
+        "Rolling back last %d migrations for plugin '%s': %s",
+        len(to_rollback),
+        plugin_name,
+        ", ".join(r.migration_id for r in to_rollback),
+    )
+
+    for row in to_rollback:
+        mig_id = row.migration_id
+        down_file = migrations_dir / f"{mig_id}.down.sql"
+        if not down_file.exists():
+            raise RuntimeError(f"Down migration not found for {plugin_name}:{mig_id} -> {down_file.name}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)